### PR TITLE
157484804 create websocket interface

### PR DESCRIPTION
**Pull request by easye**
Commit to a wallet serialization strategy.  

REST/Websocket now use a separate model of the contents of the wallet.  Moved mock behavior into model in anticipation of revision.

### CODE REVIEW

File: src/model/mock.lisp [hunk 1]
```diff
@@ -26,29 +23,10 @@
                                      (:epoch . ,epoch)
                                      (:local-epoch . ,local-epoch)
                                      (:synchronized . ,(cl-json:json-bool (= local-epoch epoch)))))))
-               (notify-clients notification))
+               (apply notify-hook notification))
```
Comment (markhdavid) [inline src/model/mock.lisp]: This cannot work: APPLY's last arg must be a list.

File: src/model/mock.lisp [hunk 2]
```diff
@@ -1,9 +1,6 @@
-(in-package :websocket/wallet)
+(in-package :model/wallet)
 
-(defvar *consensus-thread* nil)
-(defvar *consensus-clients*  nil)
-
-(defun consensus ()
+(defun mock (notify-hook)
   (let ((epoch (+ 1000 (random 10000)))
```
Comment (markhdavid) [inline src/model/mock.lisp]: I do not understand what this is doing. Can you explain the concept here a bit? I see it goes into a forever loop, but I don't really get what's going on. Is "mock" a computer term other than to mean simulate or stub something?

**Review by markhdavid — CHANGES_REQUESTED** [review]
Given that the tests are failing and there's a fatal call to APPLY, I'll reject this for now.

**Review by guitarvydas — DISMISSED** [review]
Runs my 3 tests OK.
I can't connect to ws://localhost:3145/, but that is probably OK, since I'm using a browser, not a websocket client.
I can connect to http://127.0.0.1:3140.
I see that MarkD commented as I was commenting.  I will defer to MarkD.

**Review by guitarvydas — COMMENTED** [review]
Runs my 3 tests OK.
I can't connect to ws://localhost:3145/, but that is probably OK, since I'm using a browser, not a websocket client.
I can connect to http://127.0.0.1:3140.
I see that MarkD commented as I was commenting. I will defer to MarkD.

**Comment by markhdavid** [comment]
By "tests fail" I only meant CI tests fail. I would like to know how this is to be tested though.  I could defer to Paul on that if he knows how to test.

**Comment by easye** [comment]
@guitarvydas (and possibly @markhdavid):

I have submitted patches to address your review comments on this branch.  Please review again for approval when you have a chance…

**Comment by markhdavid** [comment]
Previously, the CI was not passing. Now I see the CI is passing.  Paul stated in a private message more than an hour ago that the CI is not passing. That is not true, in fact. Plus, he has not responded to questions seeking further explanation. Accordingly, in the interest of moving forward expeditiously, I will remove my objection, and approve this.

**Review by markhdavid — DISMISSED** [review]

**Comment by guitarvydas** [comment]
screenshot 2018-06-25 17 17 31

**Comment by easye** [comment]
Closed in lieu of #278.

**Pull request closed without merging**